src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/SetAzureRmRecoveryServicesAsrReplicationProtectedItem.cs

```diff
@@ -723,6 +723,9 @@ private List<VMNicInputDetails> getNicListToUpdate(IList<VMNicDetails> vmNicList
                         var vMNicInputDetails = new VMNicInputDetails();
 
                         vMNicInputDetails.NicId = nic.NicId;
+                        vMNicInputDetails.RecoveryNicName = nic.RecoveryNicName;
```

Comment:
Can we ensure that the behavior is fine in case these values aren't provided?

Reply:
went through related asr PR (for ref https://msazure.visualstudio.com/DefaultCollection/One/_git/Azure-SiteRecovery/pullrequest/2388784). Having null values for the newly added parameters has been taken care of at service level. I have verified the same while testing.  (ex: giving just tfoName and not proving tfoRGName nd TfoRZeuseNic is a valid scenario)..

Reply:
They have been handled at service level, yes. But the point is to catch them at client level so we can alert the user earlier.

Reply:
I mean, parameters are allowed to have null values and will be populated from RPI whereever needed. Its not an error scenario to validate and raise error. Please confirm.